lib/webaction.js

```diff
@@ -116,6 +116,14 @@ async function invokeSelfAsync(params) {
 function webaction(main) {
     // actionWrapper -> webaction -> main (worker)
     return actionWrapper(async (params={}) => {
+        // Asset Compute Throttling Service metrics
+        // for quick feedback loop, store predicted run duration next to actual run duration
+        if (params && params.metrics && params.predictedRunDuration) {
+            console.log('params.metrics and params.predictedRunDuration are available');
+            params.metrics.add({
+                predictedRunDuration: params.predictedRunDuration
+            });
+        }
```

Comment:
putting it here makes it account for all the different cases:
- pipeline, non pipeline
- shellScriptWorker
- batchWorker
- regular Worker
and it works because actionWrapper adds metrics here: https://github.com/adobe/asset-compute-commons/blob/master/lib/wrap.js#L50

Reply:
Will this code will run for actions like `core`, `api-register` and I think also for custom-worker.  Which may be ok, as they won't have `params.predictedRunDuration`

Reply:
this code won't run for core and api-register and core because they use the actionWrapper directly from commons: https://git.corp.adobe.com/nui/core/blob/master/core.js#L22
but yeah the code is designed so if there is no predictedRunDuration or params.metrics, it just will ignore it